source/ceylon/file/Resource.ceylon

```diff
@@ -34,9 +34,14 @@ shared interface ExistingResource
     shared formal void writeAttribute(Attribute attribute, 
                             Object attributeValue);
     
-}
-
+    "Request that this resource be deleted when the process terminates.
 
+     Invoking [[deleteOnExit()]] does not guarantee that the resource will
+     be deleted, and may have no effect on some [[System]]s or platforms.
+     For filesystem resources on the JVM, `deleteOnExit()` delegates to
+     `java.io.File.deleteOnExit()`."
+    shared formal void deleteOnExit();
```

Comment:
Well, I could add a `variable Boolean` to track whether it's been called, and have `assign` perform double duty.

But I'm not sure I agree that it should be an attribute. We're instructing the object to add a scheduled task, not setting a property. So please confirm if you still want the change and I'll make it.